spec/smartcar/e2e/smartcar_error_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require_relative '../helpers/auth_helper'
+require_relative '../../spec_helper'
+
+RSpec.describe SmartcarError do
+  subject { Smartcar::Vehicle }
+
+  def get_vehicle_object(email = nil)
+    token_hash = AuthHelper.run_auth_flow_and_get_tokens(email, nil, ['read_odometer'])
+    vehicle_ids = Smartcar.get_vehicles(token: token_hash[:access_token])
+    subject.new(token: token_hash[:access_token], id: vehicle_ids.vehicles.first)
+  end
+
+  describe 'error' do
+    context 'with V1 requests' do
+      before(:context) do
+        Smartcar.set_api_version('1.0')
+      end
+
+      after(:context) do
+        Smartcar.set_api_version('2.0')
+      end
+      it 'should raise InvalidParameterValue error' do
+        vehicle = get_vehicle_object('smartcar@vs-000.vehicle-state-error.com')
+        expect { vehicle.odometer }.to(raise_error do |error|
```

Comment:
would you want a test for a v1.0 error without type and code? i.e. permission or authentication error